wagtail/wagtailadmin/views/pages.py

```diff
@@ -504,6 +504,25 @@ def unpublish(request, page_id):
     })
 
 
+def unshare(request, page_id):
+    page = get_object_or_404(Page, id=page_id).specific
+    if not page.permissions_for_user(request.user).can_unshare():
+        raise PermissionDenied
+
+    if request.method == 'POST':
+        page.unpublish()
+
```

Comment:
So `page.unpublish()` was wrong on line 513. It should have "unshared" the page. 

In terms of your question, publishing a page will make it appear on content. Unsharing won't work for pages that are on published.